assignments/prototype-refactor.js

```diff
@@ -7,3 +7,234 @@ Prototype Refactor
+  
+  
+    const mage = new Humanoid({
+      createdAt: new Date(),
+      dimensions: {
+        length: 2,
+        width: 1,
+        height: 1,
+      },
+      healthPoints: 5,
+      name: 'Bruce',
+      team: 'Mage Guild',
+      weapons: [
+        'Staff of Shamalama',
+      ],
+      language: 'Common Tongue',
+    });
+  
+    const swordsman = new Humanoid({
+      createdAt: new Date(),
+      dimensions: {
+        length: 2,
+        width: 2,
+        height: 2,
+      },
+      healthPoints: 15,
+      name: 'Sir Mustachio',
+      team: 'The Round Table',
+      weapons: [
+        'Giant Sword',
+        'Shield',
+      ],
+      language: 'Common Tongue',
+    });
+  
+    const archer = new Humanoid({
+      createdAt: new Date(),
+      dimensions: {
+        length: 1,
+        width: 2,
+        height: 4,
+      },
+      healthPoints: 10,
+      name: 'Lilith',
+      team: 'Forest Kingdom',
+      weapons: [
+        'Bow',
+        'Dagger',
+      ],
+      language: 'Elvish',
+    });
+  
+  // ==== NEW INSTANCES ====
+    const necromancer = new Villain({
+      createdAt: new Date(),
+      dimensions: {
+        length: 3,
+        width: 2,
+        height: 5,
+      },
+      healthPoints: 50,
+      name: 'Morag',
+      team: 'Mines',
+      weapons: [
+        'poison cloud',
+        'fireball',
+      ],
+      language: 'Orcish',
+    });
+    
+    const paladin = new Hero({
+      createdAt: new Date(),
+      dimensions: {
+        length: 2,
+        width: 5,
+        height: 1,
+      },
+      healthPoints: 50,
+      name: 'Thorin',
+      team: 'Mountain',
+      weapons: [
+        'Hammer',
+        'Axe',
+      ],
+      language: 'Dwarvish',
+    });
+  
+    console.log(mage.createdAt); // Today's date
+    console.log(archer.dimensions); // { length: 1, width: 2, height: 4 }
+    console.log(swordsman.healthPoints); // 15
+    console.log(mage.name); // Bruce
+    console.log(swordsman.team); // The Round Table
+    console.log(mage.weapons); // Staff of Shamalama
+    console.log(archer.language); // Elvish
+    console.log(archer.greet()); // Lilith offers a greeting in Elvish.
+    console.log(mage.takeDamage()); // Bruce took damage.
+    console.log(swordsman.destroy()); // Sir Mustachio was removed from the game.
+  
+    console.log(necromancer.name);
+    console.log(necromancer.healthPoints);
+    console.log(paladin.attack(necromancer));
+    console.log(necromancer.attack(paladin));
+    console.log(paladin.attack(necromancer));
+    console.log(necromancer.attack(paladin));
+    console.log(paladin.attack(necromancer));
+    console.log(necromancer.attack(paladin));
+    console.log(paladin.attack(necromancer));
+    console.log(necromancer.attack(paladin));
+    console.log(paladin.attack(necromancer));
+    console.log(necromancer.attack(paladin));
+    console.log(paladin.attack(necromancer));
+    console.log(necromancer.attack(paladin));
```

Comment:
Nice job on this, instead try using a for loop until one of them reaches 0hp and the other wins then have it end lol just an idea